plugins/tiddlywiki/multiwikiserver/modules/mws-server.js

```diff
@@ -411,11 +411,15 @@ Server.prototype.requestAuthentication = function(response) {
 };
 
 // Check if the anonymous IO configuration is set to allow both reads and writes
-Server.prototype.isAnnonIOConfigured = function() {
-	const allowReadsTiddler = this.wiki.getTiddler("$:/config/MultiWikiServer/MultiWikiServerAllowAnnonymousReads");
-	const allowWritesTiddler = this.wiki.getTiddler("$:/config/MultiWikiServer/MultiWikiServerAllowAnnonymousWrites");
-
-	return allowReadsTiddler?.fields?.text !== "undefined" && allowWritesTiddler?.fields?.text !== "undefined";
+Server.prototype.getAnonymousAccessConfig = function() {
+	const allowReadsTiddler = this.wiki.getTiddlerText("$:/config/MultiWikiServer/AllowAnonymousReads", "undefined");
+	const allowWritesTiddler = this.wiki.getTiddlerText("$:/config/MultiWikiServer/AllowAnonymousWrites", "undefined");
+
+	return {
+		allowReads: allowReadsTiddler === "yes",
+		allowWrites: allowWritesTiddler === "yes",
+		isEnabled: allowReadsTiddler !== "undefined" && allowWritesTiddler !== "undefined"
```

Comment:
I still think the config tiddlers should **not** exist as shadows. By default `isEnabled` will always be **true** because the config tiddlers exist. I think `isEnabled` should be `false` by default

Reply:
In general, I disagree, and think that the config tiddlers should exist as shadow tiddlers to make them easier to find. 

But here I think the problem is that we're struggling to understand the precise semantics of these configuration tiddlers – @webplusai could you comment?

Reply:
So there are three states this system can be in:

Unconfigured: One or both tiddlers contain "undefined" (isEnabled is false)

Configured with permissions denied: Tiddlers contain values other than "undefined", but not "yes" (isEnabled is true, but allowReads/allowWrites are false)

Configured with permissions granted: Tiddlers contain "yes" (isEnabled is true and corresponding allow flags are true)